app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt

```diff
@@ -0,0 +1,188 @@
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+import dev.chrisbanes.insetter.applyInsetter
+import eu.davidea.flexibleadapter.FlexibleAdapter
+import eu.davidea.flexibleadapter.Payload
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.databinding.ClearDatabaseControllerBinding
+import eu.kanade.tachiyomi.ui.base.controller.DialogController
+import eu.kanade.tachiyomi.ui.base.controller.FabController
+import eu.kanade.tachiyomi.ui.base.controller.NucleusController
+import eu.kanade.tachiyomi.util.system.toast
+import eu.kanade.tachiyomi.util.view.setTooltip
+import org.jsoup.internal.StringUtil.padding
+
+class ClearDatabaseController :
+    NucleusController<ClearDatabaseControllerBinding, ClearDatabasePresenter>(),
+    FlexibleAdapter.OnItemClickListener,
+    FlexibleAdapter.OnUpdateListener,
+    FabController {
+
+    private var recycler: RecyclerView? = null
+    private var adapter: FlexibleAdapter<ClearDatabaseSourceItem>? = null
+
+    private var menu: Menu? = null
+
+    var actionFab: ExtendedFloatingActionButton? = null
+    var actionFabScrollListener: RecyclerView.OnScrollListener? = null
+
+    private val selectedSources: List<ClearDatabaseSourceItem>
+        get() = adapter?.selectedPositions?.mapNotNull(adapter!!::getItem) ?: emptyList()
+
+    init {
+        setHasOptionsMenu(true)
+    }
+
+    override fun createBinding(inflater: LayoutInflater): ClearDatabaseControllerBinding {
+        return ClearDatabaseControllerBinding.inflate(inflater)
+    }
+
+    override fun createPresenter(): ClearDatabasePresenter {
+        return ClearDatabasePresenter()
+    }
+
+    override fun getTitle(): String? {
+        return activity?.getString(R.string.pref_clear_database)
+    }
+
+    override fun onViewCreated(view: View) {
+        super.onViewCreated(view)
+
+        binding.recycler.applyInsetter {
+            type(navigationBars = true) {
+                padding()
+            }
+        }
+
+        adapter = FlexibleAdapter<ClearDatabaseSourceItem>(null, this, true)
+        binding.recycler.adapter = adapter
+        binding.recycler.layoutManager = LinearLayoutManager(activity)
+        binding.recycler.setHasFixedSize(true)
+        adapter?.fastScroller = binding.fastScroller
+        recycler = binding.recycler
+    }
+
+    override fun onDestroyView(view: View) {
+        adapter = null
+        super.onDestroyView(view)
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.generic_selection, menu)
+        this.menu = menu
+        menu.apply {
+            get(0).isVisible = adapter?.itemCount ?: 0 > 0
+            get(1).isVisible = adapter?.itemCount ?: 0 > 0
+        }
```

Comment:
Can just loop over them to change the visibility accordingly
```suggestion
        menu.forEach { menuItem -> menuItem.isVisible = (adapter?.itemCount ?: 0) > 0 }
```